api/util/apps.go

```diff
@@ -117,6 +117,22 @@ func (git GitAuth) Enabled() bool {
 	return false
 }
 
+func (git GitAuth) Valid() bool {
```

Comment:
I think we could actually return either a `(bool,error)` or just an `error` to indicate which part has been empty?

Reply:
changed it to an error 👍